### ISSUE TITLE
Kubernetes Cluster is Unavailable

### ISSUE THREAD

**Issue by kjordan2001** [percona/everest#903]
I keep running into this message on my Talos cluster, but everything is up.  How can I debug what it thinks is unavailable since everything is up and operational?  It can create new database clusters, although Mongo and MySQL are always stuck in the Initializing state on version 1.3.0 even after everything is running.

**Comment by recharte** [comment]
Hi @kjordan2001, the `Kubernetes cluster is unavailable` message is potentially a kubernetes RBAC issue. However, I find this strange because you were able to create and list the database clusters.
We need to figure out which requests are returning this error. Can you please share the Everest API server logs? You can get them by running the following command:
```
kubectl logs -n everest-system -l app.kubernetes.io/component=everest
```

**Comment by kjordan2001** [comment]
[everest-log.txt](https://github.com/user-attachments/files/18027961/everest-log.txt)
Here is the log for after I refresh.  The message seems to come up when this is logged:
`{"level":"error","T":"2024-12-05T18:28:00Z","caller":"api/proxy.go:156","msg":"invalid character '\\x1f' looking for beginning of value\nfailed to transform response body"}`

**Comment by recharte** [comment]
That's very strange :sweat_smile: 
Can you show me the output of:
```
kubectl get db/mongodb-li8 -n rocketchatdb -o yaml
```

**Comment by kjordan2001** [comment]
[rocketchatdb-yaml.txt](https://github.com/user-attachments/files/18028261/rocketchatdb-yaml.txt)

**Comment by recharte** [comment]
The error message came when trying to list the backups for that DB so it seems that there might be some backups done even though the DB was stuck on initializing. They most likely failed but that still doesn't justify the strange character.
Let's have a look at the backup objects themselves to see if we spot something
```
kubectl get dbb -n rocketchatdb -o yaml
```

**Comment by recharte** [comment]
In order to understand why the DB is stuck initializing we'll also need to look into the percona-server-mongodb-operator logs.
```
kubectl logs -n rocketchatdb -l name=percona-server-mongodb-operator
```

**Comment by kjordan2001** [comment]
Output for both.
[rocketchatdb-yaml.txt](https://github.com/user-attachments/files/18028736/rocketchatdb-yaml.txt)
[mongodb-operator.txt](https://github.com/user-attachments/files/18028737/mongodb-operator.txt)

**Comment by kjordan2001** [comment]
Also for a mysql xtradb cluster that's stuck at initializing too.
[xtradb-operator.txt](https://github.com/user-attachments/files/18028851/xtradb-operator.txt)
[developmentdb-yaml.txt](https://github.com/user-attachments/files/18028862/developmentdb-yaml.txt)

**Comment by recharte** [comment]
@kjordan2001 I have a feeling that your DBs are stuck on initializing because your CNI wasn't able to provision a LoadBalancer service. Can you check this with?
```
kubectl get svc -n rocketchatdb -l app.kubernetes.io/instance=mongodb-li8
```
If you see that `mongodb-li8-rs0-0` is `<pending>` an external IP then something is wrong with your CNI plugin.

**Comment by kjordan2001** [comment]
Yes, it does say pending.  Talos uses flannel by default, I might try changing it to cilium soon.

**Comment by recharte** [comment]
I think that now I'll leave it up to you to figure out what's wrong with your cluster's network.
As a bonus tip, if the client application that will connect to the DB is deployed in the same k8s cluster as Everest, you don't need to enable external access for your DB. With external access disabled you'll have a ClusterIP service instead of a LoadBalancer service which will work around the CNI issue you're currently facing.

**Comment by efimkin** [comment]
Hi! probably that problem fixed here https://github.com/percona/everest/pull/908

**Comment by recharte** [comment]
Opened https://perconadev.atlassian.net/browse/EVEREST-1733 for internal tracking.
Thanks @efimkin for the PR :clap: 
This fix will be released in v1.5.0.

**Comment by recharte** [comment]
Due to the high impact of this bug we decided to include the fix in v1.4.0, planned to be released in a couple of weeks.
